Simplenote/Classes/SPCardViewController.swift

```diff
@@ -0,0 +1,61 @@
+import UIKit
+
+// MARK: - SPCardViewController
+//
+final class SPCardViewController: UIViewController {
+    private let viewController: UIViewController
+    private let containerView = UIView()
+
+    init(viewController: UIViewController) {
+        self.viewController = viewController
+        super.init(nibName: nil, bundle: nil)
+    }
+
+    @available(*, unavailable)
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        view.backgroundColor = .clear
+
+        addShadowView()
+        setupContainerView()
+        addChildViewController()
+    }
+}
+
+// MARK: - Private Methods
+//
+private extension SPCardViewController {
+    func addShadowView() {
+        let shadowView = SPShadowView(cornerRadius: Constants.cornerRadius,
+                                      roundedCorners: [.topLeft, .topRight])
+        view.addFillingSubview(shadowView)
+    }
+
+    func setupContainerView() {
+        view.addFillingSubview(containerView)
+
+        containerView.backgroundColor = UIColor.simplenoteBackgroundColor.withAlphaComponent(0.97)
+        containerView.layer.cornerRadius = Constants.cornerRadius
+        containerView.layer.maskedCorners = [.layerMaxXMinYCorner, .layerMinXMinYCorner]
+        containerView.layer.masksToBounds = true
+    }
+
+    func addChildViewController() {
+        addChild(viewController)
+        containerView.addFillingSubview(viewController.view)
```

Comment:
Out of curiosity, do you think it's possible to avoid using `containerView` entirely, and add the `viewController.view` as a direct subview?

Reply:
I believe it's not possible. Right now card consists of 2 subviews: 
* shadow view, which draws shadow on the outside of bounds rect
* content view, which has rounded corners and clips inside content, so that child view controller doesn't need to worry about any of those

If we would add child view controller directly to the main view, we would need to apply rounded corners and clip subviews. This would clip and remove shadow.